source/gameengine/Rasterizer/RAS_ICanvas.cpp

```diff
@@ -90,14 +90,16 @@ void save_screenshot_thread_func(TaskPool *__restrict UNUSED(pool), void *taskda
 	/* create and save imbuf */
 	ImBuf *ibuf = IMB_allocImBuf(task->dumpsx, task->dumpsy, 24, 0);
 	ibuf->rect = task->dumprect;
-
+	
 	BKE_imbuf_write_as(ibuf, task->path, task->im_format, false);
-
+	task->path = NULL;
```

Comment:
Indeed path need to be set to NULL. But as you can see later we alloc the name with
`	char *path = (char *)MEM_mallocN(FILE_MAX, "screenshot-path");`

So you have also to free the path:
`
MEM_freeN(task->path);
task->path = NULL;
`